packages-web/maps/src/components/__tests__/GoogleMap.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { shallow, ShallowWrapper } from "enzyme";
+import { createElement } from "react";
+import { GoogleMap, GoogleMapsProps } from "../GoogleMap";
+import { initialize } from "@googlemaps/jest-mocks";
+
+describe("Google maps", () => {
+    const defaultProps: GoogleMapsProps = {
+        autoZoom: true,
+        height: 75,
+        heightUnit: "pixels",
+        scriptsLoaded: true,
+        optionDrag: true,
+        optionScroll: true,
+        optionZoomControl: true,
+        zoomLevel: 10,
+        width: 50,
+        widthUnit: "percentage",
+        mapStyles: "",
+        style: {},
+        currentLocation: undefined,
+        optionStreetView: false,
+        locations: [],
+        className: "",
+        fullScreenControl: false,
+        mapsToken: "",
+        mapTypeControl: false,
+        rotateControl: false,
+        showCurrentLocation: false
+    };
+
+    beforeEach(() => {
+        initialize();
+    });
+
+    const renderGoogleMap = (props: GoogleMapsProps): ShallowWrapper<GoogleMapsProps, any> =>
+        shallow(createElement(GoogleMap, props));
+
+    it("renders a map with right structure", () => {
+        const googleMaps = renderGoogleMap(defaultProps);
+        googleMaps.setProps({
+            heightUnit: "percentageOfWidth",
+            widthUnit: "pixels"
+        });
+
+        expect(googleMaps).toMatchSnapshot();
+    });
+
+    it("renders a map with pixels renders structure correctly", () => {
+        const googleMaps = renderGoogleMap(defaultProps);
+        googleMaps.setProps({
+            heightUnit: "pixels",
+            widthUnit: "pixels"
+        });
+
+        expect(googleMaps).toMatchSnapshot();
+    });
+
+    it("renders a map with percentage of width and height units renders the structure correctly", () => {
+        const googleMaps = renderGoogleMap(defaultProps);
+        googleMaps.setProps({
+            heightUnit: "percentageOfWidth",
+            widthUnit: "percentage"
+        });
+
+        expect(googleMaps).toMatchSnapshot();
+    });
+
+    it("renders a map with percentage of parent units renders the structure correctly", () => {
+        const googleMaps = renderGoogleMap(defaultProps);
+        googleMaps.setProps({
+            heightUnit: "percentageOfParent",
+            widthUnit: "percentage"
+        });
+
+        expect(googleMaps).toMatchSnapshot();
+    });
+
```

Comment:
Done

Reply:
You used snapshots for these tests, but if you check `snap` files - snapshots don't include markers, because they are added by GoogleMaps and not through React.